### PR TITLE
Filter effect collection by expression

### PR DESCRIPTION
**Pull request by iskandr**
* added `filter_transcript_expression` and `filter_gene_expression` methods to `EffectCollection`
* fixed `summary_string` on `EffectCollection`
* renamed `summary_string` -> `detailed_string`
* changed __str__ method to print collection elements 
* added a `short_string` method which doesn't print collection elements
* added unit tests for common Collection methods

[Review on Reviewable](https://reviewable.io/reviews/hammerlab/varcode/67)

### CODE REVIEW

File: varcode/collection.py [hunk 1]
```diff
@@ -62,15 +63,21 @@ def __eq__(self, other):
...
         """
         return self.__class__(
-            elements=new_elements,
+            new_elements,
             filename=self.filename)
 
+    def filter(self, filter_fn):
+        return self.clone_with_new_elements([
+            element
+            for element in self.element
```
Comment (arahuja) [inline varcode/collection.py]: `self.elements`?

File: varcode/collection.py [hunk 2]
```diff
@@ -62,15 +63,21 @@ def __eq__(self, other):
             len(self) == len(other) and
             all(x == y for (x, y) in zip(self._elements, other._elements)))
 
-    def _clone_metadata(self, new_elements):
+    def clone_with_new_elements(self, new_elements):
         """
         Create copy of VariantCollection with same metadata but possibly
```
Comment (arahuja) [inline varcode/collection.py]: Not new code, but this is not specific to `VariantCollection` anymore

File: varcode/effect_collection.py [hunk 3]
```diff
@@ -79,13 +151,17 @@ def summary_string(self):
...
+                best = variant_effects.top_priority_effect()
                 lines.append("  Highest Priority Effect: %s" % best)
         return "\n".join(lines)
 
+    def show(self):
+        """
+        Print effects in this collection, grouped by mutation and
+        sub-grouped by gene
+        """
+        print(self.summary_string())
```
Comment (arahuja) [inline varcode/effect_collection.py]: `summary_string()` is gone now?

File: varcode/effects.py [hunk 4]
```diff
@@ -124,7 +125,7 @@ def __init__(self, variant, transcript):
     def __str__(self):
         return "%s(variant=%s, transcript=%s)" % (
             self.__class__.__name__,
-            self.variant.short_description(),
+            self.variant.compact_string,
```
Comment (arahuja) [inline varcode/effects.py]: Does this exist?

File: varcode/collection.py (hunk 1 above)
Comment (iskandr) [inline varcode/collection.py]: :+1:

File: varcode/collection.py (hunk 2 above)
Comment (iskandr) [inline varcode/collection.py]: :+1:

File: varcode/effect_collection.py (hunk 3 above)
Comment (iskandr) [inline varcode/effect_collection.py]: I switched the `__str__` method to actually just print the first 100 collection elements, so don't need these any more.

**Pull request merged**